Automattic/pocket-casts-ios
# Referrals: IAP setup

| 📘 Part of: #2083  |  
|:---:|

Fixes #2184  


This PR updates the IAP code to use a new subscription id specific for referrals.

The logic to process a Referral purchase will be the following:
- If the user never had a subscription before, we will offer them a Introductory Offer with 2 months free
- If the user had a subscription before, but it's not active we will offer them a new subscription with a promotional offer of two 2 months. **NOTE**: This PR is still no implementing the promotional offer because server is not ready yet.
- After the user uses the referral offer he should never be allowed anymore to have any Trial Offer or the Referral Offer again.

| 1 | 2 - Intro Offer |
| - | - |
| ![Simulator Screenshot - iPhone 16 Pro - 2024-10-07 at 13 24 00](https://github.com/user-attachments/assets/e4727af3-450f-449e-8bc8-bfcf91fe582c) | ![Simulator Screenshot - iPhone 16 Pro - 2024-10-07 at 13 24 07](https://github.com/user-attachments/assets/6d8ab990-1090-4f10-a478-055d28525d5d) 

## To test

- Start the app on the device
- Ensure that you the referrals FF enabled
- Ensure that you have Testflight - Sandbox account so you can do fake IAP calls. Ping me if you need help with this.

### Setup:
- Get a referral link from a user with a Plus or Patron account. You can use your current paid account
- Go to Profile
- Tap on Gift icon on the top left
- Tap on Send Guest Pass
- Copy the referral link and save it to the Reminders app
- You will use this link on the following steps
- Log out from the Plus/Patron account

### Scenario 1 - User that never had a subscription before

- The easiest way to setup this is to create a new user account
- Tap on the Referral link obtained on the Setup on the Reminders app
- Check that The Claim referral Guest pass page is opened 
- Tap on Claim Pass
- Check that the IAP flow start and you see the referral Intro Offer
- Purchase it

## Checklist
